src/api/app/controllers/concerns/rescue_authorization_handler.rb

```diff
@@ -3,40 +3,47 @@ module RescueAuthorizationHandler
 
   included do # rubocop:disable Metrics/BlockLength
     rescue_from Pundit::NotAuthorizedError do |exception|
-      if RoutesHelper::APIMatcher.matches?(request)
-        render_error status: 403,
-                     errorcode: authorization_errorcode(exception),
-                     message: authorization_message(exception)
-      else
-        respond_to do |format|
-          format.js { render json: { error: authorization_message(exception) }, status: 400 }
-          format.any do
-            flash[:error] = authorization_message(exception)
-            redirect_path = unauthorized_redirect_path(exception)
-            if redirect_path
-              redirect_to(redirect_path)
-            else
-              redirect_back(fallback_location: root_path)
-            end
+      respond_to do |format|
+        format.html do
+          flash[:error] = authorization_message(exception)
+          redirect_path = unauthorized_redirect_path(exception)
+          if redirect_path
+            redirect_to(redirect_path)
+          else
+            redirect_back(fallback_location: root_path)
           end
         end
+        format.json { render json: { errorcode: authorization_errorcode(exception), summary: authorization_message(exception) }, status: 403 }
+        # Consider everything else an XML request...
+        format.any do
+          @errorcode = authorization_errorcode(exception)
+          @summary = authorization_message(exception)
+          render template: 'status', status: 403, formats: [:xml]
+        end
       end
     end
 
     private
 
     def action_for_exception(exception)
-      case exception.query.to_s.chop
+      action = exception.query || 'show?'
+      action = action.to_s.chop
+
+      case action
       when 'index' then 'list'
       when 'show' then 'view'
       when 'new' then 'create'
       when 'destroy' then 'delete'
-      else exception.query.to_s.chop
+      else action
```

Comment:
@hennevogel just wanted to point to the else part since its not needed. You defaul to `show` anyway. But it triggers rubocop. So little suggestion:

Reply:
```
ACTION_FOR_EXCEPTION = {
    'index' => 'list',
    'show' => 'view',
    'new' => 'create',
    'destroy' => 'delete'
  }.freeze
```

Reply:
Adding this to the module

Reply:
and then 
```
def action_for_exception(exception)
      action = exception.query || 'show?'
      action = action.to_s.chop

      ACTION_FOR_EXCEPTION[action]
    end
```

Reply:
The else part is returning the action name. It triggers rubocop where, not here?

Reply:
There are more actions that those, somewhere you will need to return the ones we are not translating.

Reply:
alright :+1: , sorry for the noice